### ISSUE TITLE
Mypy doesn't detect unreachable if

### ISSUE THREAD

**Issue by hackaugusto** [python/mypy#8286]
* Are you reporting a bug, or opening a feature request?
Feature request (?)

* Please insert below the code you are checking with mypy,
  or a mock-up repro if the source is private. We would appreciate
  if you try to simplify your case to a minimal repro.
```python
class A:
    pass

if type(1) in (A,):
    print("this should give a warning")

if type(1) == A:
    print("this works")
```
* What is the actual behavior/output?
```
t.py:7: error: Non-overlapping equality check (left operand type: "Type[int]", right operand type: "Type[A]")
Found 1 error in 1 file (checked 1 source file)
```
* What is the behavior/output you expect?
Two errors, one for each impossible branch.
* What are the versions of mypy and Python you are using?
  Do you see the same issue after installing mypy from Git master?
Tried with `0.720`, `0.761` and `0.770+dev.974a58d3cb3029044b481f9e3154fa3a185cf0f3`
* What are the mypy flags you are using? (For example --strict-optional)
Tried with combinations of `--strict`, `--strict-equality`, and `--warn-unreachable`

**Comment by JukkaL** [comment]
It looks like strict equality container checks don't work with fixed-length tuples. Another example:

```py
# mypy: strict-equality
from typing import Tuple

t1: Tuple[str, ...]
t2: Tuple[str, str]

if 1 in t1:  # Non-overlapping container check (expected)
    pass

if 1 in t2:  # No error (unexpected!)
    pass
```
